pkg/applicationserver/io/pubsub/provider/mqtt/driver.go

```diff
@@ -264,10 +260,8 @@ func (s *subscription) Close() error {
 		return nil
 	}
 	token := s.client.Unsubscribe(s.topic)
-	if !token.WaitTimeout(timeout) {
-		return errUnsubscribeFailed.WithCause(context.DeadlineExceeded)
-	} else if token.Error() != nil {
-		return errUnsubscribeFailed.WithCause(token.Error())
+	if err := waitToken(s.ctx, token); err != nil {
```

Comment:
I can imagine that the caller calls `Close()` when the context is already canceled, immediately returning `waitToken()`. I wonder what the value of `ctx` is in `Close()`?

Should we use a new context with short timeout here, or are we really sure we can use the context?

Reply:
`s.ctx` is the context that was used to create the subscription, which is the context that was used to open the provider connection. You are correct that on a pub/sub cancellation, `waitToken` will return immediately. 

But if that context is closed/expired - the connection is getting closed as well, so is there any value in waiting ? Waiting an unsubscribe here just to kill the connection afterwards seems a bit pointless. 

We can of course extend `context.Background()`, but do we get much out of it ?

Reply:
OK fine to use `s.ctx` then.